lib/pbench/server/api/resources/server_configuration.py

```diff
@@ -113,40 +113,74 @@ def _put(self, params: ApiParams, _) -> Response:
         }
         """
 
-        try:
-            key = params.uri["key"]
-            try:
-                value = params.query["value"]
-            except KeyError:
-                try:
-                    value = params.body["value"]
-                except KeyError:
+        # First, we check for the ways to set an individual config setting by
+        # naming the config key on the URI and specifying a value either with
+        # the "value" either as a query parameter or in the JSON request body.
+        #
+        # We'll complain about JSON request body parameters that are "shadowed"
+        # by the URI or query parameter and might represent client confusion.
+        keydup = set()
+        key = params.uri.get("key")
+        if key:
+            # If we have a key in the URL, the we need a "value" for it, which
+            # we can take either from a query parameter or from the JSON
+            # request payload.
+            if params.body and list(params.body.keys()) != ["value"]:
+                keydup.update(params.body.keys())
+            value = params.query.get("value")
+            if value:
+                if params.body:
+                    keydup.update(params.body.keys())
```

Comment:
Two different focuses, but probably could be cleaned up... ☹️

Reply:
On reflection ... I think that I want to complain if we get

```
PUT /server/configuration/k?value=v
{
    "value": "v"
}
```

... because that might mean they expected the body value...

... but probably not if we get

```
PUT /server/configuration/k
{
    "value": "v",
    "something": "else"
}
```
... I'm basically replacing normal schema keyword/JSON validation here after dropping the `"config"` key and expecting first-level key name. But I designed the schema validator to be forgiving essentially about version skew by ignoring unrecognized values instead of complaining. Therefore no reason to complain *here* about extra values.